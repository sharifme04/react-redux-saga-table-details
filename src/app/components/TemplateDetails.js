import React from "react";
import PropTypes from "prop-types";

const TemplateDetails = ({ match, templates, history }) => {
  const details = templates?._data?.filter(
    // eslint-disable-next-line eqeqeq
    (template) => template.templateId == match.params.templateId
  )[0];

  return (
    <div className="container">
      <h2>Template Details</h2>
      <hr />
      <div className="card text-center">
        <div className="card-header">Template</div>
        <div className="card-body">
          <h5 className="card-title">ID: {details?.templateId}</h5>
          <h6 className="card-title">Name: {details?.name}</h6>
          <h6 className="card-title">Partner ID: {details?.partnerId}</h6>
          <p className="card-text">
            <b>optional Documents (Top Level Domain) : </b>
            {details?.optionalDocuments?.assetType?.conditions?.topLevelDomain}
          </p>
          <p className="card-text">
            <b>optional Documents (Email Address) : </b>
            {details?.optionalDocuments?.assetType?.values?.emailAddress
              ? "Available"
              : "Not available"}
          </p>
          <p className="card-text">
            <b>Required Documents (Top Level Domain) : </b>
            {details?.requiredDocuments?.assetType?.conditions?.topLevelDomain}
          </p>
          <p className="card-text">
            <b>Required Documents (Email Address) : </b>
            {details?.requiredDocuments?.assetType?.values?.emailAddress
              ? "Required"
              : "Not Required"}
          </p>
          <p>
            <b>Type:</b> {details?.type}
          </p>
          <p>
            <b>Valid From:</b> {details?.validFrom}
          </p>
          <p>
            <b>Valid To:</b> {details?.validTo}
          </p>
          <button
            className="btn btn-secondary"
            onClick={() => history.push("/")}
          >
            Go Back
          </button>
        </div>
        <div className="card-footer text-muted">
          Version: {details?.version}
        </div>
      </div>
    </div>
  );
};

TemplateDetails.propTypes = {
  templates: PropTypes.shape({
    status: PropTypes.number,
    message: PropTypes.string,
    _data: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string,
        itemplateId: PropTypes.string,
        partnerId: PropTypes.string,
        type: PropTypes.string,
        version: PropTypes.number,
        validFrom: PropTypes.string,
        validTo: PropTypes.string,
        optionalDocuments: PropTypes.shape({
          assetType: PropTypes.shape({
            conditions: PropTypes.shape({
              topLevelDomain: PropTypes.string,
            }),
            values: PropTypes.shape({
              emailAddress: PropTypes.bool,
            }),
          }),
        }),
        requiredDocuments: PropTypes.shape({
          assetType: PropTypes.shape({
            conditions: PropTypes.shape({
              topLevelDomain: PropTypes.string,
            }),
            values: PropTypes.shape({
              emailAddress: PropTypes.bool,
            }),
          }),
        }),
      })
    ),
  }),
};

export default TemplateDetails;
