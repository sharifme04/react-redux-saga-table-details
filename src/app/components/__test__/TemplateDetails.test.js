import React from "react";
import { shallow } from "enzyme";
import TemplateDetails from "../TemplateDetails";

const template = {
  conditions: null,
  name: "basic_user",
  optionalDocuments: {
    assetType: {
      conditions: { topLevelDomain: ".edu" },
      values: { emailAddress: true },
    },
  },
  partnerId: "59c25c00-78cd-11ea-ac9b-bf680aaa6d4b",
  requiredDocuments: {
    assetType: {
      conditions: { topLevelDomain: ".edu" },
      values: { emailAddress: true },
    },
  },
  templateId: "487c5b70-9516-11ea-bbfa-ed3b490e43f0",
  type: "basic",
  validFrom: "04-05-2020",
  validTo: "21-05-2020",
  version: 2,
};

describe("<TemplateDetails />", () => {
  test("renders", () => {
    const component = shallow(<TemplateDetails details={template} />);
    expect(component).toMatchSnapshot();
  });

  test("return the default array when there is no data to map through", () => {
    const component = shallow(<TemplateDetails />);
    expect(component).toMatchSnapshot();
  });
  test("does not break without value", () => {
    const component = shallow(<TemplateDetails />);
    expect(component.find("h2")).toHaveLength(1);
    expect(component.find("h5")).toHaveLength(1);
    expect(component.find("h6")).toHaveLength(2);
    expect(component.find("p")).toHaveLength(7);
    expect(component.find("button")).toHaveLength(1);
  });

  test("does not break with an empty object", () => {
    const component = shallow(<TemplateDetails details={{}} />);
    expect(component.find("h2")).toHaveLength(1);
    expect(component.find("h5")).toHaveLength(1);
    expect(component.find("h6")).toHaveLength(2);
    expect(component.find("p")).toHaveLength(7);
    expect(component.find("button")).toHaveLength(1);
  });
});
