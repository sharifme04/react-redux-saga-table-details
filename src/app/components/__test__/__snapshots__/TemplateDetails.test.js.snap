// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemplateDetails /> renders 1`] = `
<div
  className="container"
>
  <h2>
    Template Details
  </h2>
  <hr />
  <div
    className="card text-center"
  >
    <div
      className="card-header"
    >
      Template
    </div>
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        ID: 
      </h5>
      <h6
        className="card-title"
      >
        Name: 
      </h6>
      <h6
        className="card-title"
      >
        Partner ID: 
      </h6>
      <p
        className="card-text"
      >
        <b>
          optional Documents (Top Level Domain) : 
        </b>
      </p>
      <p
        className="card-text"
      >
        <b>
          optional Documents (Email Address) : 
        </b>
        Not available
      </p>
      <p
        className="card-text"
      >
        <b>
          Required Documents (Top Level Domain) : 
        </b>
      </p>
      <p
        className="card-text"
      >
        <b>
          Required Documents (Email Address) : 
        </b>
        Not Required
      </p>
      <p>
        <b>
          Type:
        </b>
         
      </p>
      <p>
        <b>
          Valid From:
        </b>
         
      </p>
      <p>
        <b>
          Valid To:
        </b>
         
      </p>
      <button
        className="btn btn-secondary"
        onClick={[Function]}
      >
        Go Back
      </button>
    </div>
    <div
      className="card-footer text-muted"
    >
      Version: 
    </div>
  </div>
</div>
`;

exports[`<TemplateDetails /> return the default array when there is no data to map through 1`] = `
<div
  className="container"
>
  <h2>
    Template Details
  </h2>
  <hr />
  <div
    className="card text-center"
  >
    <div
      className="card-header"
    >
      Template
    </div>
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        ID: 
      </h5>
      <h6
        className="card-title"
      >
        Name: 
      </h6>
      <h6
        className="card-title"
      >
        Partner ID: 
      </h6>
      <p
        className="card-text"
      >
        <b>
          optional Documents (Top Level Domain) : 
        </b>
      </p>
      <p
        className="card-text"
      >
        <b>
          optional Documents (Email Address) : 
        </b>
        Not available
      </p>
      <p
        className="card-text"
      >
        <b>
          Required Documents (Top Level Domain) : 
        </b>
      </p>
      <p
        className="card-text"
      >
        <b>
          Required Documents (Email Address) : 
        </b>
        Not Required
      </p>
      <p>
        <b>
          Type:
        </b>
         
      </p>
      <p>
        <b>
          Valid From:
        </b>
         
      </p>
      <p>
        <b>
          Valid To:
        </b>
         
      </p>
      <button
        className="btn btn-secondary"
        onClick={[Function]}
      >
        Go Back
      </button>
    </div>
    <div
      className="card-footer text-muted"
    >
      Version: 
    </div>
  </div>
</div>
`;
